sendto_silhouette.py

```diff
@@ -567,6 +606,18 @@ def effect(self):
 
         self.logEnvironment()
 
+        # Auto detect mark to document corner and mark to mark distance from Regmark layer (Quality of life improvement)
+        # Generated Regmarks from inkscape silhouette embeds ids so we can automatically derive offset calculations from it
+        self.detect_regmark()
+
+        # Registration Mark Selection/Calcuation
+        # Note that any user defined values will overrride any auto calcuated values
+        reg_origin_X = self.options.regoriginx or self.doc_reg_x
+        reg_origin_Y = self.options.regoriginy or self.doc_reg_y
+        reg_width = self.options.regwidth or self.doc_reg_width or self.svg.to_dimensional(self.svg.viewport_width, "mm") - reg_origin_X * 2
+        reg_length = self.options.reglength or self.doc_reg_length or self.svg.to_dimensional(self.svg.viewport_height, "mm") - reg_origin_Y * 2
+        self.report(f"Using Registration Mark:: mark distance from document: Left={reg_origin_X}mm, Top={reg_origin_Y}mm; mark to mark distance: X={reg_width}mm, Y={reg_length}mm;", 'log')
```

Comment:
the very last thing that bothers me: can we pack this part into the function as well?
i.e. include the logic of this 5 lines also withing detect_regmark()

Reply:
I get the other one, but this seems like diminishing return kind of refactoring/cleanup. But anyway I've added it in. Just waiting for the checks to pass and let's see how it goes!